[JENKINS-70781] hudson.util.ProcessTreeTest#processProperlyKilledUnix fails on macOS

<!-- ### Imported Jira references for easier searching -->
<!-- [jira_issue_key=JENKINS-70781] -->
<!-- [jira_issue_type=Bug] -->
<!-- [reporter=notmyfault] -->
<!-- [author=notmyfault] -->
<!-- [jira_component=core] -->
<!-- [jira_label=help-wanted] -->
<!-- [jira_issues_importer_version=1.2.0] -->

<p>Building core via <tt>mvn clean verify -Dtest=hudson.util.ProcessTreeTest#processProperlyKilledUnix</tt> emits the following error:</p>

<pre>
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 8.286 s <<< FAILURE! - in hudson.util.ProcessTreeTest
[ERROR] hudson.util.ProcessTreeTest.processProperlyKilledUnix  Time elapsed: 8.261 s  <<< FAILURE!
java.lang.AssertionError: 

Expected: not a string containing <span class="code-quote">"sleep 100000"</span>
     but: was <span class="code-quote">"Legacy code started <span class="code-keyword">this</span> job.  No cause information is available\nRunning as SYSTEM\nBuilding in workspace /Users/alex/github/jenkins/test/target/j h16377702109887518028/workspace/test1\n[test1] $ /bin/sh -xe /Users/alex/github/jenkins/test/target/jenkins991816571209404711.sh\n+ ps -ef\n+ grep sleep\n  501   864   756   0 Sat09AM ?? 0:00.00 sleep 31536000\n  501 48340     1   0 11:53PM ?? 0:00.01 sleep 100000\n  501 61234     1   0 10:33AM ?? 0:00.01 sleep 100000\n  501 61240 61238   0 10:33AM ?? 0:00.00 grep sleep\nFinished: SUCCESS\n"</span>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.jvnet.hudson.test.JenkinsRule.assertLogNotContains(JenkinsRule.java:1513)
	at hudson.util.ProcessTreeTest.processProperlyKilledUnix(ProcessTreeTest.java:90)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:608)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:1589)
</pre>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/notmyfault.png" title="notmyfault's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=notmyfault">notmyfault</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-70781" target="_blank">hudson.util.ProcessTreeTest#processProperlyKilledUnix fails on macOS</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>component(s)</b>: core
<li><b>label(s)</b>: help-wanted
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 0
<li><b>watchers</b>: 2
<li><b>imported</b>: 2025-11-24
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Building core via <tt>mvn clean verify -Dtest=hudson.util.ProcessTreeTest#processProperlyKilledUnix</tt> emits the following error:</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 8.286 s &lt;&lt;&lt; FAILURE! - in hudson.util.ProcessTreeTest
[ERROR] hudson.util.ProcessTreeTest.processProperlyKilledUnix  Time elapsed: 8.261 s  &lt;&lt;&lt; FAILURE!
java.lang.AssertionError: 

Expected: not a string containing <span class="code-quote">"sleep 100000"</span>
     but: was <span class="code-quote">"Legacy code started <span class="code-keyword">this</span> job.  No cause information is available\nRunning as SYSTEM\nBuilding in workspace /Users/alex/github/jenkins/test/target/j h16377702109887518028/workspace/test1\n[test1] $ /bin/sh -xe /Users/alex/github/jenkins/test/target/jenkins991816571209404711.sh\n+ ps -ef\n+ grep sleep\n  501   864   756   0 Sat09AM ??         0:00.00 sleep 31536000\n  501 48340     1   0 11:53PM ??         0:00.01 sleep 100000\n  501 61234     1   0 10:33AM ??         0:00.01 sleep 100000\n  501 61240 61238   0 10:33AM ??         0:00.00 grep sleep\nFinished: SUCCESS\n"</span>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.jvnet.hudson.test.JenkinsRule.assertLogNotContains(JenkinsRule.java:1513)
	at hudson.util.ProcessTreeTest.processProperlyKilledUnix(ProcessTreeTest.java:90)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.jvnet.hudson.test.JenkinsRule$1.evaluate(JenkinsRule.java:608)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.<span class="code-object">Thread</span>.run(<span class="code-object">Thread</span>.java:1589)
</pre>
</div></div></pre>
</details>
</details>
<ul><li><i>environment</i>: <code>Building core with maven 3.9.0 on eclipse temurin 19.0.2 on macOS 13.2.1</code></li></ul>

--- a/test/src/test/java/hudson/util/ProcessTreeTest.java
+++ b/test/src/test/java/hudson/util/ProcessTreeTest.java
@@ -88,13 +88,14 @@ class ProcessTreeTest {
         FreeStyleProject sleepProject = j.createFreeStyleProject();
         FreeStyleProject processJob = j.createFreeStyleProject();
 
-        sleepProject.getBuildersList().add(new Shell("nohup sleep 100000 &"));
+        int sleepTime = 90000 + java.util.concurrent.ThreadLocalRandom.current().nextInt(20000);
+        sleepProject.getBuildersList().add(new Shell("nohup sleep " + sleepTime + " &"));
 
         j.buildAndAssertSuccess(sleepProject);
 
         processJob.getBuildersList().add(new Shell("ps -ef | grep sleep"));
 
-        j.assertLogNotContains("sleep 100000", processJob.scheduleBuild2(0).get());
+        j.assertLogNotContains("sleep " + sleepTime, processJob.scheduleBuild2(0).get());
     }
 
     @Test

